Source/Dafny/Compilers/Compiler-js.cs

```diff
@@ -2386,12 +2386,32 @@ protected class ClassWriter : IClassWriter {
         CreateNoWindow = true,
         UseShellExecute = false,
         RedirectStandardInput = true,
-        RedirectStandardOutput = false,
-        RedirectStandardError = false,
+        RedirectStandardOutput = true,
+        RedirectStandardError = true,
       };
 
       try {
         using var nodeProcess = Process.Start(psi);
+        nodeProcess.BeginErrorReadLine();
+        nodeProcess.BeginOutputReadLine();
+
+        void ProcessErrorData(object sender, DataReceivedEventArgs e) {
+          if (!nodeProcess.HasExited) {
+            Console.Error.WriteLine(e.Data);
+          } else {
+            Console.Error.Write(e.Data);
+          }
+        }
+
+        void ProcessOutputData(object sender, DataReceivedEventArgs e) {
+          if (!nodeProcess.HasExited) {
+            Console.Out.WriteLine(e.Data);
+          } else {
+            Console.Out.Write(e.Data);
+          }
+        }
+        nodeProcess.ErrorDataReceived += ProcessErrorData;
```

Comment:
Move the registration of the event-handlers before the asynchronous setup `BeginErrorReadLine` to not miss any process output due to the race condition.